contactql/parser_test.go

```diff
@@ -111,6 +116,58 @@ func TestParseQuery(t *testing.T) {
+		{`created_on != 20-02-2020`, `created_on != "20-02-2020"`, "", envs.RedactionPolicyNone},
+		{`tel != 02352`, `tel != 02352`, "", envs.RedactionPolicyNone},
+		{`urn != 02352`, `urn != 02352`, "", envs.RedactionPolicyNone},
+		{`age != 18`, `age != 18`, "", envs.RedactionPolicyNone},
+		{`gender != male`, `gender != "male"`, "", envs.RedactionPolicyNone},
+		{`dob != 20-02-2020`, `dob != "20-02-2020"`, "", envs.RedactionPolicyNone},
+		{`state != Pichincha`, `state != "Pichincha"`, "", envs.RedactionPolicyNone},
+
+		// ~ only supported for name and URNs
+		{`id ~ 02352`, ``, "contains conditions can only be used with name or URN values", envs.RedactionPolicyNone},
+		{`name ~ felix`, `name ~ "felix"`, "", envs.RedactionPolicyNone},
+		{`language ~ eng`, ``, "contains conditions can only be used with name or URN values", envs.RedactionPolicyNone},
+		{`group ~ porters`, ``, "contains conditions can only be used with name or URN values", envs.RedactionPolicyNone},
+		{`created_on ~ 2018`, ``, "contains conditions can only be used with name or URN values", envs.RedactionPolicyNone},
+		{`tel ~ 02352`, `tel ~ 02352`, "", envs.RedactionPolicyNone},
+		{`urn ~ 02352`, `urn ~ 02352`, "", envs.RedactionPolicyNone},
+		{`age ~ 18`, ``, "contains conditions can only be used with name or URN values", envs.RedactionPolicyNone},
+		{`gender ~ mal`, ``, "contains conditions can only be used with name or URN values", envs.RedactionPolicyNone},
+		{`dob ~ 20-02-2020`, ``, "contains conditions can only be used with name or URN values", envs.RedactionPolicyNone},
+		{`state ~ Pichincha`, ``, "contains conditions can only be used with name or URN values", envs.RedactionPolicyNone},
+
+		// > >= < <= only supported for numeric or date fields
+		{`id > 02352`, ``, "comparisons with > can only be used with date and number fields", envs.RedactionPolicyNone},
+		{`name > felix`, ``, "comparisons with > can only be used with date and number fields", envs.RedactionPolicyNone},
+		{`language > eng`, ``, "comparisons with > can only be used with date and number fields", envs.RedactionPolicyNone},
+		{`group > reporters`, ``, "comparisons with > can only be used with date and number fields", envs.RedactionPolicyNone},
+		{`created_on > 20-02-2020`, `created_on > "20-02-2020"`, "", envs.RedactionPolicyNone},
+		{`tel > 02352`, ``, "comparisons with > can only be used with date and number fields", envs.RedactionPolicyNone},
+		{`urn > 02352`, ``, "comparisons with > can only be used with date and number fields", envs.RedactionPolicyNone},
+		{`age > 18`, `age > 18`, "", envs.RedactionPolicyNone},
+		{`gender > male`, ``, "comparisons with > can only be used with date and number fields", envs.RedactionPolicyNone},
+		{`dob > 20-02-2020`, `dob > "20-02-2020"`, "", envs.RedactionPolicyNone},
+		{`state > Pichincha`, ``, "comparisons with > can only be used with date and number fields", envs.RedactionPolicyNone},
```

Comment:
@nicpottier these new tests above look consistent with recent tweaks on the ES side?

Reply:
I think so, but as I see this I wonder if we should just move the `search` package into goflow so these things are closer together. It doesn't really depend on anything within mailroom and having them both in goflow would allow for more natural testing etc.. King of like the providers for email etc, having a standard implementation of contactql to elastic in goflow seems totally reasonable.

Reply:
Yeah that doesn't sound terrible. I've been wracking my brain thinking of ways to enforce consistency between these two evaluator implementations.. and if we moved the ES stuff into goflow.. I think we could have some tests that assert consistency